workflow.go

```diff
@@ -418,13 +423,14 @@ func NewTimer(ctx Context, d time.Duration) Future {
 	})
 	if t != nil {
 		Go(ctx, func(ctx Context) {
-			if ctx.Done() == nil {
-				return // not cancellable.
+			if ctxDone := ctx.Done(); ctxDone != nil {
+				NewSelector(ctx).AddReceive(ctxDone, func(c Channel, more bool) {
+					// We will cancel the timer either it is explicit cancellation (or) we are closed.
```

Comment:
Can you add unit test for activity/timer/child workflow? you can use the API to query the stack and check of the co-routine still exist.